enhancements/upstream-downstream-split.md

```diff
@@ -0,0 +1,228 @@
+- Coordinate versioning across OLM projects
+- Support the previous [n-2] releases
+
+#### Preserve Upstream Velocity
+- Add collaborators who are not Red Hat employees, when warranted, without risk to openshift
+- Insulate upstream development from the downstream release cadence and branch policies
+- Allow merges to development branches at any time
+- Run tests independent of the OpenShift CI infrastructure
+- Upstream branching strategy is not restricted by compatibility with downstream branching strategy
+
+#### Preserve Downstream Velocity
+- Downstream can’t be blocked by upstream decisions or indecision
+- Shepherding changes from upstream to downstream should not require undue toil: Possibly have this process fully automated
+- Retain compatibility with existing OpenShift release process and tooling
+- Minimize disruption to regular development during migration period
+
+### Non-Goals
+
+- Zero OpenShift-specific code in upstream from day 1
+- Changes to downstream conformance testing
+- Changes to issue and design proposal process
+- Changes affecting repositories other than `operator-framework/operator-lifecycle-manager`, `operator-framework/operator-registry` or `operator-framework/api`
+- Test enhancements (e.g. parallelization)
+
+## Proposal
+
+### User Stories
+
+#### Unblocked development for downstream
+As an OLM developer, I want to make changes to downstream code without being blocked by upstream.
+
+#### Independent development for upstream
+As an upstream contributor, I want to make contributions without depending on downstream.
+
+#### Sync downstream with upstream
+As an OLM developer, I want to bring upstream changes to downstream branches painlessly.
+
+#### Bring changes to upstream
+As an OLM developer, I want to bring specific downstream changes to upstream for community review and switch to the upstream version of the changes if and when they are accepted.
+
+### Implementation Details
+
+The downstream product is composed of the upstream projects plus downstream-only additions and modifications. Since downstream is expected to write its own entrypoints and extension packages, code from upstream dependencies is isolated from the accompanying downstream code.
+
+Downstream necessity still allows for direct modifications to copies of upstream dependencies, although ideally most problems can be resolved without doing so to reduce divergence from upstream. Backports will be supported for the previous [n-2] releases ([[n-5] cycles for releases with extended update support](https://access.redhat.com/support/policy/updates/openshift#ocp4_phases)).
+
+The `operator-framework/operator-lifecycle-manager`, `operator-framework/operator-registry` and `operator-framework/api` repositories will serve as the upstream repositories while a new repository `openshift/operator-lifecycle-manager` will be the downstream monorepo for all of the listed upstream repositories (and any that may be added in future).
```

Comment:
Is there any specific reason for making it a monorepo? Why can't they continue to be individual repos?

Reply:
If we have individual repos, then we'd need to keep the one-to-one mapping between upstream and downstream repos in future every time upstream decides to split/merge/add/remove repos